robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -217,11 +217,11 @@ def execute(self, userdata=None) -> str:
         # Remove pose from ED as we are holding the object in the gripper
         self.robot.ed.update_entity(uuid=grab_entity.uuid, remove_pose=True)
 
-        arm.wait_for_motion_done(cancel=True)
-
         # Carrying pose
         arm.send_joint_goal('carrying_pose', timeout=0.0)
 
+        arm.wait_for_motion_done(cancel=True)
+
```

Comment:
The robot can start driving while going to the carrying pose. Often this goes well and it makes the robot faster. I prefer the old way of doing this.